test/cli_test.py

```diff
@@ -421,7 +421,14 @@ def test_run_parse_args_entrypoint(servicer, set_env_client, test_dir):
         assert len(servicer.function_call_inputs) == 0
 
     res = run_cli_command(["run", f"{app_file.as_posix()}::unparseable_annot", "--i=20"], expected_exit_code=1)
-    assert "Parameter `i` has unparseable annotation: typing.Union[int, str]" in str(res.exception)
+
+    if sys.version_info >= (3, 14):
+        # Python 3.14 normalizes the type to `int | str`.
+        union_str = "int | str"
+    else:
+        union_str = "typing.Union[int, str]"
```

Comment:
Not 100% sure I'm understanding but is this going to confuse people if they literally write `Union[a, b]` and we tell them that their `a | b` annotation is unparseable? ("But ... I didn't write that!")

Might be farfetched that 3.14 users are still spelling it as `Union[...]` but are there other normalizations that might be happening here? Wondering if we want to slightly rephrase the error message so it doesn't look like we are claiming the normalized representation is literally in the user code. I might be overthinking this though.